### ISSUE TITLE
CoreOS does not work on GCE

### ISSUE THREAD

**Issue by christus02** [kubernetes/kops#7149]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

Version 1.12.1

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

v1.14.2

**3. What cloud provider are you using?**

Google Cloud (GCE)

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

```
kops create cluster cic-coreos-kubenet.k8s.local --state gs://<my google bucket>/ --image coreos-cloud/coreos-stable-2079-5-1-v20190606 --networking kubenet --cloud gce --zones asia-south1-a --project <my google project> --yes
```

**5. What happened after the commands executed?**

VM instances were created, but cluster did not come up. I logged into to the Master's nodes shell and found that even `kubectl` was not working.

`kubectl get nodes` also failed from my kubectl client

```
# kubectl  get nodes 
The connection to the server <IP> was refused - did you specify the right host or port?
```

**6. What did you expect to happen?**

The cluster should have been created and `kubectl get nodes` would have given the nodes in the cluster 

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

```
apiVersion: kops/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: 2019-06-16T20:37:06Z
  name: cic-coreos-kubenet.k8s.local
spec:
  api:
    loadBalancer:
      type: Public
  authorization:
    rbac: {}
  channel: stable
  cloudProvider: gce
  configBase: <MASKED>
  etcdClusters:
  - cpuRequest: 200m
    etcdMembers:
    - instanceGroup: master-asia-south1-a
      name: a
    memoryRequest: 100Mi
    name: main
  - cpuRequest: 100m
    etcdMembers:
    - instanceGroup: master-asia-south1-a
      name: a
    memoryRequest: 100Mi
    name: events
  iam:
    allowContainerRegistry: true
    legacy: false
  kubelet:
    anonymousAuth: false
  kubernetesApiAccess:
  - 0.0.0.0/0
  kubernetesVersion: 1.12.8
  masterPublicName: api.cic-coreos-kubenet.k8s.local
  networking:
    kubenet: {}
  nonMasqueradeCIDR: 100.64.0.0/10
  project: netscaler-networking-k8
  sshAccess:
  - 0.0.0.0/0
  subnets:
  - name: asia-south1
    region: asia-south1
    type: Public
  topology:
    dns:
      type: Public
    masters: public
    nodes: public

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2019-06-16T20:37:08Z
  labels:
    kops.k8s.io/cluster: cic-coreos-kubenet.k8s.local
  name: master-asia-south1-a
spec:
  image: coreos-cloud/coreos-stable-2079-5-1-v20190606
  machineType: n1-standard-1
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-asia-south1-a
  role: Master
  subnets:
  - asia-south1
  zones:
  - asia-south1-a

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2019-06-16T20:37:10Z
  labels:
    kops.k8s.io/cluster: cic-coreos-kubenet.k8s.local
  name: nodes
spec:
  image: coreos-cloud/coreos-stable-2079-5-1-v20190606
  machineType: n1-standard-2
  maxSize: 2
  minSize: 2
  nodeLabels:
    kops.k8s.io/instancegroup: nodes
  role: Node
  subnets:
  - asia-south1
  zones:
  - asia-south1-a
```

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

**9. Anything else do we need to know?**

**Comment by srikiz** [comment]
I see a similar issue with the latest master only. I see things working OK with release-1.13. Do you have any updates if this is working for you, please let me know.

**Comment by juergenthomann** [comment]
I tested it with the current CoreOS version and kops 1.13 and it does not work for me as well.

The problem is as far as I found out that CoreOS on GCE runs the startup script in a container called oem-gce and this prevents most of the involved systemd commands to work. For example, the start/restarts commands are ignored as systemd detects a chroot.

I would expect it to be broken for a long time as @justinsb reported a problem with CoreOS on GCE already a year ago at https://github.com/coreos/bugs/issues/2502
In https://github.com/coreos/bugs/issues/2558 it is stated that startup-script is deprecated and that the way forward would be Container Linux Configs or Ignition.
